collectors/osd.go

```diff
@@ -687,10 +738,29 @@ func (o *OSDCollector) collectOSDScrubState(ch chan<- prometheus.Metric) error {
 	return nil
 }
 
+func (o *OSDCollector) collectPGRecoveryState(ch chan<- prometheus.Metric) error {
+	for _, pg := range o.pgDumpBrief {
+		if strings.Contains(pg.State, "recovering") {
```

Comment:
This looks good! Is the idea to add backfill as a separate PR? I am fine either ways!

Reply:
Yep, I would like to make another PR for backfill, as the changes in PR have already been many.